plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,167 @@
+ */
+function od_optimization_detective_rest_api_test(): array {
+	$result = array(
+		'label'       => __( 'The Optimization Detective REST API endpoint is functional.', 'optimization-detective' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Optimization Detective', 'optimization-detective' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			__( 'Your site can send and receive URL metrics via the Optimization Detective REST API endpoint.', 'optimization-detective' )
+		),
+		'actions'     => '',
+		'test'        => 'optimization_detective_rest_api',
+	);
+
+	$rest_url = get_rest_url( null, OD_REST_API_NAMESPACE . OD_URL_METRICS_ROUTE );
+	$response = wp_remote_post(
+		$rest_url,
+		array(
+			'headers'   => array( 'Content-Type' => 'application/json' ),
+			'sslverify' => false,
+		)
+	);
+
+	if ( is_wp_error( $response ) ) {
+		$result['status']      = 'recommended';
+		$result['label']       = __( 'Error accessing the Optimization Detective REST API endpoint', 'optimization-detective' );
+		$result['description'] = esc_html__( 'There was an issue reaching the Optimization Detective REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		$info                  = array(
+			'error_message' => $result['description'],
+			'error_code'    => $response->get_error_code(),
+			'available'     => false,
+		);
+	} else {
+		$status_code = wp_remote_retrieve_response_code( $response );
+		$data        = json_decode( wp_remote_retrieve_body( $response ), true );
+		$info        = array(
+			'status_code'   => $status_code,
+			'available'     => false,
+			'error_message' => '',
+		);
+
+		if (
+			400 === $status_code &&
+			isset( $data['data']['params'] ) &&
+			is_array( $data['data']['params'] ) &&
+			count( $data['data']['params'] ) > 0
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the Optimization Detective REST API endpoint.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'To collect URL metrics, the Optimization Detective REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' );
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Access to the Optimization Detective REST API endpoint was denied.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'Access was denied because the user does not have the necessary capabilities. Please check user roles and capabilities, as all users should have access to the Optimization Detective REST API endpoint.', 'optimization-detective' );
```

Comment:
If this happens most likely, the permission required for the endpoint was changed, e.g. by another plugin. So maybe we can rephrase the recommendation slightly:
```suggestion
			$result['description'] = esc_html__( 'Access was denied, possibly because the user does not have the necessary capabilities. This might be because the required capabilities to access the REST API were modified or because the REST API is blocked due to server configuration.', 'optimization-detective' );
```

Reply:
I've fleshed this out in d9577451a